src/ctapipe/core/tool.py

```diff
@@ -452,18 +452,21 @@ def run(self, argv=None, raises=False):
                 if raises:
                     raise
             except SystemExit as err:
-                if raises:
-                    raise  # do not re-intercept in tests
-                else:
-                    exit_status = err.code
-                    self.log.exception(
-                        "Caught SystemExit with exit code %s", exit_status
-                    )
-                    Provenance().finish_activity(
-                        activity_name=self.name,
-                        status="error",
-                        exit_code=exit_status,
-                    )
+                exit_status = err.code
+                if (
+                    exit_status != 0
+                ):  # Do nothing if SystemExit was called with the exit code 0 (e.g. with -h option)
```

Comment:
If you place the comment in the line above or below the if, ruff won't split the if in this weird way